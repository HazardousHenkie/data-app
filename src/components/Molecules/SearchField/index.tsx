import React, { useState, useEffect } from 'react'

import { useTranslation } from 'react-i18next'

import InputFieldStyled from './styledComponents'

interface SearchFieldProps {
    setValue: React.Dispatch<React.SetStateAction<string>>
}

const SearchField: React.FC<SearchFieldProps> = ({ setValue }) => {
    const [inputValue, setInputValue] = useState('')
    // suspense makes the map not so we will have handle the not ready state here
    const { t, ready } = useTranslation('searchField', { useSuspense: false })

    useEffect(() => {
        const timer = setTimeout(() => {
            setValue(inputValue)
        }, 500)

        return () => clearTimeout(timer)
    }, [inputValue, setValue])

    const handleOnChange = (
        event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
    ) => {
        setInputValue(event.target.value)
    }

    return (
        <form>
            {ready && (
                <InputFieldStyled
                    placeholder={t('searchField:inputLabel', 'search')}
                    onChange={handleOnChange}
                />
            )}
        </form>
    )
}

export default SearchField
