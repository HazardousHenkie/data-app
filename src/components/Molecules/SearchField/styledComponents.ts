import styled from 'styled-components'

import InputField from 'components/Atoms/InputField'

const InputFieldStyled = styled(InputField)`
    width: 100%;

    input {
        width: 100%;
    }
`

export default InputFieldStyled
