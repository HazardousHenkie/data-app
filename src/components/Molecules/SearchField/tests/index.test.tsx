import React from 'react'
import renderer from 'react-test-renderer'

import MockingAppComponent from 'utils/testing/MockingAppComponent'
import SearchField from '../index'

describe('<SearchField />', () => {
    it('should render like snapshot', () => {
        const component = renderer.create(
            <MockingAppComponent>
                <SearchField setValue={() => {}} />
            </MockingAppComponent>
        )

        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
