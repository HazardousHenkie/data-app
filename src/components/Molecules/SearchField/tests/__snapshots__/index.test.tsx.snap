// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchField /> should render like snapshot 1`] = `<form />`;
